lib/dpl/provider/deis.rb

```diff
@@ -44,6 +44,31 @@ def setup_git_ssh(path, key_path)
         unless context.shell "./deis git:remote --app=#{option(:app)}"
           error 'Adding git remote failed.'
         end
+
+        wait_for_git_access
+      end
+
+      def wait_for_git_access()
+        retry_count=0
+        max_retries=30
+
+        #Parse out the git remote host and port
+        git_remote=`cat .git/config  | grep -A1 '[remote] "deis"' | tail -n 1 | sed 's/.*url = //'`
```

Comment:
It's bordering on style, but there is no reason to pipe `cat` into `grep`; `grep -A1 '[remote] "deis"' .git/config` should be sufficient.